tests/Feature/MovieTest.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+namespace Tests\Feature;
+
+use Tests\TestCase;
+use App\Models\User;
+use App\Models\Movie;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class MovieTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    public function an_admin_can_see_movies_view()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->get('/movies')
+            ->assertOk();
+    }
+
+    /** @test */
+    public function a_user_can_see_movies_view()
+    {
+        $user = User::factory()->create([
+            'is_admin' => false,
+        ]);
+
+        $this->actingAs($user)
+            ->get('/movies')
+            ->assertOk();
+    }
+
+    /** @test */
+    public function an_admin_can_access_create_movie_view()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->get('/movies/create')
+            ->assertOk();
+    }
+
+    /** @test */
+    public function a_user_cannot_access_create_movie_view()
+    {
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($user)
+            ->get('/movies/create')
+            ->assertForbidden();
```

Comment:
Now that I think about it, it would probably be better to create a policy instead of middleware.
Then you can add permissions to the user check the [Laravel docs-Policies]( https://laravel.com/docs/5.2/authorization#policies) and see what you think 😄

Reply:
See [Laravel: Difference between middleware and policy](https://stackoverflow.com/questions/35019292/laravel-difference-between-route-middleware-and-policy)

Reply:
Ahh okay Vick, for now I'll edit the test to assert the status and then we can make further changes once we talk to Steve about the middleware :)